libcontainer/dmz/cloned_binary_linux.go

```diff
@@ -0,0 +1,195 @@
+//go:build linux
+// +build linux
+
+package dmz
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+
+	"github.com/opencontainers/runc/libcontainer/system"
+)
+
+type SealFunc func(**os.File) error
+
+var (
+	_ SealFunc = sealMemfd
+	_ SealFunc = sealFile
+)
+
+const baseMemfdSeals = unix.F_SEAL_SEAL | unix.F_SEAL_SHRINK | unix.F_SEAL_GROW | unix.F_SEAL_WRITE
+
+func sealMemfd(f **os.File) error {
+	if err := (*f).Chmod(0o511); err != nil {
+		return fmt.Errorf("chmod memfd: %w", err)
+	}
+	// Try to set the newer memfd sealing flags, but we ignore
+	// errors because they are not needed and we want to continue
+	// to work on older kernels.
+	fd := (*f).Fd()
+	// F_SEAL_FUTURE_WRITE -- Linux 5.1
+	_, _ = unix.FcntlInt(fd, unix.F_ADD_SEALS, unix.F_SEAL_FUTURE_WRITE)
+	// F_SEAL_EXEC -- Linux 6.3
+	const F_SEAL_EXEC = 0x20 //nolint:revive // this matches the unix.* name
```

Comment:
Wanted to add this constant to x/sys/unix, but apparently F_SEAL_* are taken from glibc include file (rather than the kernel one), so I sent a patch to glibc to add it: https://sourceware.org/pipermail/libc-alpha/2023-August/151125.html

Reply:
NB: this was merged and will be available in glibc v2.39. Guess it will make its way into x/sys/unix eventually.